ext/app/electron/config.ts

```diff
@@ -0,0 +1,206 @@
+      confValue = confValue[segment]
+      if (confValue === undefined) {
+        break
+      }
+    }
+    let envValue = process.env[envKey]
+    if (envValue === undefined) {
+      if (confValue === undefined) {
+        log.info(`Neither ${confKey} nor $${envKey} is specified, using default value ${defaultValue}`)
+        process.env[envKey] = defaultValue
+        return
+      } else {
+        // envvar is undefined but config has it
+        if (!validator(confValue)) {
+          log.warn(`${confKey} has invalid value ${confValue}, using default value ${defaultValue}`)
+          process.env[envKey] = defaultValue
+        } else {
+          process.env[envKey] = confValue
+        }
+      }
+    } else {
+      // envvar is defined, ignore config
+      if (confValue !== undefined) {
+        log.warn(`${confKey} is overridden by $${envKey}`)
+      }
+      if (!validator(envValue)) {
+        log.warn(`$${envKey} has invalid value ${envValue}, using default value ${defaultValue}`)
+        process.env[envKey] = defaultValue
+      }
+    }
+  }
+}
+
+
+function loadConfigFile(filename: string = DEFAULT_CONFIG_FILE) {
+  let config: Config
+  try {
+    let configBuffer = fse.readFileSync(filename)
+    config = new Config(ini.parse(configBuffer.toString("utf8")))
+  } catch (err) {
+    log.warn(`Failed to read configuration file: ${err}`)
+    config = new Config({})
+  }
+  // Section: login
+  config.apply(
+    "login.username",
+    "GRIST_DEFAULT_USERNAME",
+    NO_VALIDATION,
+    getUsername()
+  )
+  config.apply(
+    "login.email",
+    "GRIST_DEFAULT_EMAIL",
+    NO_VALIDATION,
+    getEmail()
+  )
+  // Section: server
+  config.apply(
+    "server.listen",
+    "GRIST_HOST",
+    NO_VALIDATION,
+    "localhost"
+  )
+  config.apply(
+    "server.port",
+    "GRIST_PORT",
+    (portstr) => {
+      if (! /^\d+$/.test(portstr)) {
+        return false
+      }
+      let port = parseInt(portstr)
+      return port > 0 && port < 65536
+    },
+    "_RANDOM"
+  )
+  config.apply(
+    "server.auth",
+    "GRIST_DESKTOP_AUTH",
+    (auth) => ["strict", "none", "mixed"].includes(auth),
+    "strict"
+  )
+  // Section: sandbox
+  config.apply(
+    "sandbox.flavor",
+    "GRIST_SANDBOX_FLAVOR",
+    (flavor) => ["pyodide", "unsandboxed", "gvisor", "macSandboxExec"].includes(flavor),
+    "pyodide"
+  )
+  // Section: storage
+  config.apply(
+    "storage.instance",
+    "GRIST_INST_DIR",
+    NO_VALIDATION,
+    APPDATA_DIR
+  )
+  config.apply(
+    "storage.documents",
+    "GRIST_DATA_DIR",
+    NO_VALIDATION,
+    electron.app.getPath("documents")
+  )
+  config.apply(
+    "storage.plugins",
+    "GRIST_USER_ROOT",
+    NO_VALIDATION,
+    path.join(electron.app.getPath("home"), ".grist")
+  )
+  config.apply(
+    "storage.homedb",
+    "TYPEORM_DATABASE",
+    NO_VALIDATION,
+    path.join(APPDATA_DIR, "home.sqlite3")
+  )
+
+  const homedb_location = path.parse((process.env.TYPEORM_DATABASE as string)).dir
```

Comment:
all names should be javascript style. For variable names that  would be thisKindOfStyle.